cmd/access/access_node_builder.go

```diff
@@ -0,0 +1,195 @@
+package main
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/cmd"
+	"github.com/onflow/flow-go/cmd/build"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/module/local"
+	"github.com/onflow/flow-go/network"
+	jsoncodec "github.com/onflow/flow-go/network/codec/json"
+	"github.com/onflow/flow-go/network/p2p"
+	"github.com/onflow/flow-go/network/topology"
+	"github.com/onflow/flow-go/network/validator"
+)
+
+// AccessNodeBuilder is initializes and runs an Access node either as a staked node or an unstaked node
+// It is composed of the FlowNodeBuilder
+type AccessNodeBuilder struct {
```

Comment:
Given the way an unstaked access node (and a staked access node which is supporting unstaked access nodes) need to be bootstrapped, a custom NodeBuilder is needed. The FlowNodeBuilder does most of the initialization but it needs be customized to bring up the unstaked network and skip certain check for the unstaked access node.